server/test/unit/com/thoughtworks/go/server/service/EntityHashingServiceTest.java

```diff
@@ -1,58 +1,66 @@
+import com.thoughtworks.go.util.CachedDigestUtils;
+import com.thoughtworks.go.util.ConfigElementImplementationRegistryMother;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
 
 import static org.hamcrest.CoreMatchers.is;
 import static org.junit.Assert.assertThat;
-import static org.mockito.Matchers.any;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.verifyZeroInteractions;
 import static org.mockito.Mockito.when;
 
 public class EntityHashingServiceTest {
-    private EntityDigest entityDigest;
     private GoConfigService goConfigService;
     private GoCache goCache;
     private EntityHashingService entityHashingService;
+    private ConfigCache configCache;
+    private ConfigElementImplementationRegistry registry;
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
 
     @Before
     public void setUp() {
-        this.entityDigest = mock(EntityDigest.class);
         this.goConfigService = mock(GoConfigService.class);
         this.goCache = mock(GoCache.class);
-        this.entityHashingService = new EntityHashingService(this.goConfigService, this.goCache);
+        this.configCache = new ConfigCache();
+        this.registry = ConfigElementImplementationRegistryMother.withNoPlugins();
+        this.entityHashingService = new EntityHashingService(this.goConfigService, this.goCache, configCache, registry);
 
-        this.entityHashingService.initializeWith(entityDigest);
     }
 
     @Test
-    public void shouldGenerateMD5ForAPipelineConfig() {
-        PipelineConfig pipelineConfig = mock(PipelineConfig.class);
-
-        when(goConfigService.pipelineConfigNamed(any(CaseInsensitiveString.class))).thenReturn(pipelineConfig);
-        when(entityDigest.md5ForPipeline(pipelineConfig)).thenReturn("pipeline_config_md5");
+    public void shouldThrowAnExceptionWhenObjectIsNull() {
+        thrown.expect(NullPointerException.class);
+        entityHashingService.md5ForEntity(null, null);
+    }
 
-        String md5ForPipeline = entityHashingService.md5ForPipelineConfig("P1");
+    @Test
+    public void shouldComputeTheMD5OfAGivenXmlPartialGeneratedFromAnObject() {
+        PipelineConfig pipelineConfig = PipelineConfigMother.pipelineConfig("P1");
+        String xml = new MagicalGoConfigXmlWriter(configCache, registry).toXmlPartial(pipelineConfig);
 
-        assertThat(md5ForPipeline, is("pipeline_config_md5"));
-        verify(goCache).put("GO_PIPELINE_CONFIGS_ETAGS_CACHE", "p1", "pipeline_config_md5");
+        assertThat(entityHashingService.md5ForEntity(pipelineConfig, "P1"), is(CachedDigestUtils.md5Hex(xml)));
     }
 
     @Test
     public void shouldReturnCachedMD5IfPresent() {
-        when(goCache.get("GO_PIPELINE_CONFIGS_ETAGS_CACHE", "p1")).thenReturn("pipeline_config_md5");
+        when(goCache.get("GO_ETAG_CACHE", "p1")).thenReturn("pipeline_config_md5");
```

Comment:
This would involve changing the access specifier from private to protected. Is that necessary for the test's sake?